src/main/java/com/gregtechceu/gtceu/api/item/IGTTool.java

```diff
@@ -911,20 +910,7 @@ T> LazyOptional<T> getCapability(@NotNull Capability<T> capability,
 
     default boolean definition$isCorrectToolForDrops(ItemStack stack, BlockState state) {
         if (stack.getItem() instanceof IGTTool gtTool) {
-            boolean isCorrectToolType = gtTool.getToolClasses(stack).stream()
-                    .anyMatch(type -> type.harvestTags.stream().anyMatch(state::is));
-            if (!isCorrectToolType) {
-                return false;
-            }
-
-            final int totalLevel = gtTool.getTotalHarvestLevel(stack);
-            List<Tier> tiers = TierSortingRegistry.getSortedTiers().stream()
-                    .filter(tier -> tier.getLevel() == totalLevel)
-                    .toList();
-            Tier tier = !tiers.isEmpty() ? tiers.get(tiers.size() - 1) : null;
-
-            if (tier == null) return false;
-            return TierSortingRegistry.isCorrectTierForDrops(tier, state);
```

Comment:
I do believe removing these extra checks will cause issues.

Reply:
There is a similar check in isToolEffectively (). This puzzles me

Reply:
I believe that method doesn't check the tool type.

Reply:
If you're referring to GTToolType, I think this [code](https://github.com/GregTechCEu/GregTech-Modern/blob/36da46adf3bb783376295a38dfb62be302bef06a/src/main/java/com/gregtechceu/gtceu/api/item/tool/ToolHelper.java#L617) has been checked

Reply:
If you're referring to GTToolType, I think this [code](https://github.com/GregTechCEu/GregTech-Modern/blob/36da46adf3bb783376295a38dfb62be302bef06a/src/main/java/com/gregtechceu/gtceu/api/item/tool/ToolHelper.java#L617) has performed inspections

Reply:
well, no, it doesn't do the
```java
final int totalLevel = gtTool.getTotalHarvestLevel(stack);
            List<Tier> tiers = TierSortingRegistry.getSortedTiers().stream()
                    .filter(tier -> tier.getLevel() == totalLevel)
                    .toList();
            Tier tier = !tiers.isEmpty() ? tiers.get(tiers.size() - 1) : null;

            if (tier == null) return false;
            return TierSortingRegistry.isCorrectTierForDrops(tier, state);
```

Reply:
https://github.com/GregTechCEu/GregTech-Modern/blob/36da46adf3bb783376295a38dfb62be302bef06a/src/main/java/com/gregtechceu/gtceu/api/item/tool/ToolHelper.java#L618 Please take a look at this method

Reply:
yes, I did, it does it slightly differently.

Reply:
also that one seems to just be flat out broken because it only calls dropWhile which drops until the first valid instance.
it also doesn't get the last one in the list, but any one.

Reply:
Yes, I am comparing the differences in logic. Thank you for your reminder. :)

Reply:
In my opinion, if the corresponding Tier exists, this method always returns a Tier whose level is greater than or equal to the input level. This is acceptable in my opinion.

Reply:
no, it needs to return the highest valid tier, as otherwise a tool with X mining level might not be able to mine a block with the same mining level but sorted slightly higher.